builder.md

```diff
@@ -1,57 +1,142 @@
+with base16 theming somehow, in a way that makes sense for their intended
+usecase.
 
-Each template repository should have a templates folder containing a config.yaml and any needed mustache template files.
+Template maintainers SHOULD provide built versions (with all existing scheme)
+so the end user doesn't need to be aware of the builder.
 
-- `/templates/*.mustache` - A template file (there may be multiples of these)
-- `/templates/config.yaml` - A template configuration file
+## Interface
 
-## Workflow
-The first thing a builder needs to do is parse all the scheme files from the schemes repository (as defined in the [file guidelines](https://github.com/chriskempson/base16/blob/master/file.md)). All files matching the pattern `*.yaml` should be loaded from the root of the schemes repository.
+All base16 builders MUST provide a single feature: building a template using 1 or more schemes.
 
-When building a target template repository, a base16 builder should first clear out any old output. Next, for all templates defined in the template repo's config file (located at `/templates/config.yaml`), the builder should iterate through all the defined schemes and output matching files. The built filename should look like `[output-dir]/base16-[slug][extension]`, where the slug is taken from the scheme filename made lowercase with spaces replaced with dashes and both the extension and output-dir are taken from `/template/config.yaml`.
+### CLI
 
-In the case where schemes share the same slug, a builder will overwrite files previously generated from the template. Should this happen, a builder should show warning messages listing the overwritten files.
+It is REQUIRED that that this functionality is exposed as binary CLI executable
+with the exact interface described below.
+
+The binary name SHOULD contain `base16`, but is otherwise left as a choice to
+the author.
+
+To be fully compliant, a builder CLI interface MUST NOT include any other
+feature, option, argument, or deviance from the expected interface and
+behaviour of the program.
+
+<!-- TODO: For convenience, we make a manpage and set of tests available. All
+compliant builders MUST fully conform to these two. -->
+
+The CLI interface MUST work without relying on any network connection.
+
+```bash
+base16 TEMPLATES-DIRECTORY SCHEME-FILE ...
+```
+
+`TEMPLATES-DIRETORY` being a directory containing a `config.yaml`, as well as at
+least one `.mustache` template. These are usually named `templates` on all
+base16 template repositories.
+
+`SCHEME-FILE` being one or more `.yaml` scheme file. The main source for these
+is the [base16-schemes](https://github.com/base16-project/base16-schemes)
+repository.
+
+The program MAY offer a single `--debug | -d` option for increasing log
+verbosity. These are considered implementation detail and each author SHOULD
+implement them as they see fit.
+
+### Library
+
+The compliant base16 builder software MAY also expose a software library other
```

Comment:
What do you mean by this? Can you give an example?

Reply:
Sure. 

Suppose you created a python builder. It probably has some kind of class for representing schemes or templates, possibly method or functions for color conversion or the template replacement itself.

It could be useful for other python programs, for example, a GUI that uses base16 to build some sort of theme. The creator could avoid writing code to implement the base16 standard (parsing schemes, messing with the template, etc) if there was some kind of library.

If the base16 is already very general purpose, why not expose your classes/functions/methods as a library for other python software to use?